dotnet/src/Connectors/Connectors.OpenAI/TextToImage/AzureOpenAITextToImageService.cs

```diff
@@ -3,215 +3,136 @@
 using Microsoft.SemanticKernel.Http;
-using Microsoft.SemanticKernel.Services;
 using Microsoft.SemanticKernel.TextToImage;
 
 namespace Microsoft.SemanticKernel.Connectors.OpenAI;
 
 /// <summary>
-/// Azure OpenAI text to image service
+/// Azure OpenAI Image generation
 /// <see herf="https://learn.microsoft.com/en-us/azure/cognitive-services/openai/reference#image-generation" />
 /// </summary>
 [Experimental("SKEXP0012")]
 public sealed class AzureOpenAITextToImageService : ITextToImageService
 {
-    private readonly OpenAITextToImageClientCore _core;
-
-    /// <summary>
-    /// Generation Image Operation path
-    /// </summary>
-    private const string GenerationImageOperation = "openai/images/generations:submit";
-
-    /// <summary>
-    /// Get Image Operation path
-    /// </summary>
-    private const string GetImageOperation = "openai/operations/images";
-
-    /// <summary>
-    /// Azure OpenAI REST API endpoint
-    /// </summary>
-    private readonly string _endpoint;
-
-    /// <summary>
-    /// Azure OpenAI API key
-    /// </summary>
-    private readonly string _apiKey;
-
-    /// <summary>
-    /// Maximum number of attempts to retrieve the text to image operation result.
-    /// </summary>
-    private readonly int _maxRetryCount;
-
-    /// <summary>
-    /// Azure OpenAI Endpoint ApiVersion
-    /// </summary>
-    private readonly string _apiVersion;
-
+    private readonly OpenAIClient _client;
+    private readonly ILogger _logger;
+    private readonly string _deploymentName;
     /// <summary>
-    /// Create a new instance of Azure OpenAI text to image service
+    /// Create a new instance of Azure OpenAI image generation service
     /// </summary>
-    /// <param name="endpoint">Azure OpenAI deployment URL, see https://learn.microsoft.com/azure/cognitive-services/openai/quickstart</param>
-    /// <param name="apiKey">Azure OpenAI API key, see https://learn.microsoft.com/azure/cognitive-services/openai/quickstart</param>
-    /// <param name="modelId">Azure OpenAI model id, see https://learn.microsoft.com/azure/cognitive-services/openai/how-to/create-resource</param>
+    /// <param name="deploymentName">Deployment name identifier</param>
+    /// <param name="endpoint">Azure OpenAI deployment URL</param>
+    /// <param name="apiKey">Azure OpenAI API key</param>
     /// <param name="httpClient">Custom <see cref="HttpClient"/> for HTTP requests.</param>
     /// <param name="loggerFactory">The ILoggerFactory used to create a logger for logging. If null, no logging will be performed.</param>
-    /// <param name="maxRetryCount"> Maximum number of attempts to retrieve the text to image operation result.</param>
     /// <param name="apiVersion">Azure OpenAI Endpoint ApiVersion</param>
     public AzureOpenAITextToImageService(
-        string? endpoint,
+        string deploymentName,
+        string endpoint,
         string apiKey,
-        string? modelId = null,
```

Comment:
We should keep the `modelId` for telemetry purposes in the short term.

Longer term if Azure OpenAI supports both Dalle 3&3 then we will want to allow developers to specify which model to use and use the service selector to select the correct service.

Reply:
Brought it back